### ISSUE TITLE
Inconsistent data format for API `listParentDSTrio`

### ISSUE THREAD

**Issue by amaltaro** [dmwm/dbs2go#9]
While validating the WMCore DBS3Reader, as from this discussion:
https://github.com/dmwm/WMCore/issues/10842

tests for the `listParentDSTrio` API - currently only used for the StepChain parentage fix thread - are failing and the reason for that is a different output data format between the Python and Go-base servers.

It can be checked with something like:
```
dbs = DbsApi(dbsURL)
dbs.listParentDSTrio(dataset='/Cosmics/ComissioningHI-PromptReco-v1/RECO')
```

where the python-based server responds with data like:
```
[{'36092526': [[180851, 2], [180851, 1]]}, {'36092560': [[180841, 14],...
```

while the go-based server response is:
```
{36092526: [[180851, 2], [180851, 1]], 36092560: [[180841, 23], [180841, 2],...
```

In short, 2 problems:
* output is not encapsulated in a list object
* key for the dictionary is not a string

**Comment by vkuznet** [comment]
@amaltaro thanks for spotting this, it is silly mistake in DBSClient. I fixed it in this PR https://github.com/dmwm/DBSClient/pull/67 But I doubt we should go back to put keys as string. Those are parent ids and as such are not string by definition.

Let me know how to proceed. I can cut new DBSClient release to fix output list object issue, but I don't know how we should deal with keys. Can you point out where exactly they used in WMCore code? We can check if there are anything related to data-type of the key.

**Comment by amaltaro** [comment]
@vkuznet Valentin, with the client fix that you provided, our DBS3Reader.getParentDatasetTrio method no longer crashes. It's still not clear to me whether we will be able to write those file ids back - using a write API - in the format of an integer. But it's likely the case in the database schema, so I'm pretty confident with this.

Once you merge that and release a new dbs3-client version, I think we can close this out.
I'm reluctant in getting yet another deployment of reqmgr2 with this new client, even though this would be the safest approach...

**Comment by vkuznet** [comment]
new release 4.0.7 is available on PyPi. I'm closing this ticket.
